src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -705,11 +705,10 @@ def __call__(self):
                 raise ex
 
         # Check process.source exists
-        self.logger.info("Debug: Self.process")
-        self.logger.info(dir(self.process))
         if getattr(self.process, "source", None) is None and getattr(self.process, "_Process__source", None) is None:
-            msg = "Error in CMSSW PSet: process is missing attribute 'source'"
-            msg += " or process.source is defined with None value."
+            msg = "Error in CMSSW PSet: process object is either missing or has "
+            msg += "None value for attributes 'source' and '_Process__source'. "
+            msg += f"Details of process object are: {dir(self.process)}"
```

Comment:
I think this is dangerous. I can see the point of printing it only for debug purposes, on an error. But as one can see from the previous error message one of the situations where this can be triggered is when the whole process object is missing or completely broken. Hence trying to print it through `dir()` without checking for its existence, to me sounds a little bit dangerous or misleading: e.g. :

```
>>> pprint(dir(None))
['__bool__',
 '__class__',
 '__delattr__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__le__',
 '__lt__',
 '__ne__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__']
``` 

So the call to `dir()` will return something which would mask the real error, in the case of completely missing `process` object.  

p.s. Also please add some reformatting of the log message with `pformat`,  for it is a quite long content dumped in an extremely  unreadable single line.

Thanks!

Reply:
`self.process` is initialized in the contructor. So it will be there, even if it is a None object.
The pformat idea is good though, as those extremely large lines are painful to look at. I am updating it in a bit.